File: src/de/sciencecomputing/venuside/psi/ContextFile.java
package de.sciencecomputing.venuside.psi;

import com.intellij.extapi.psi.PsiFileBase;
import com.intellij.openapi.fileTypes.FileType;
import com.intellij.psi.FileViewProvider;
import org.jetbrains.annotations.NotNull;
import de.sciencecomputing.venuside.ContextFileType;
import de.sciencecomputing.venuside.ContextLanguage;

public class ContextFile extends PsiFileBase {
    public ContextFile(@NotNull FileViewProvider viewProvider) {
        super(viewProvider, ContextLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public FileType getFileType() {
        return ContextFileType.INSTANCE;
    }

    @Override
    public String toString() {
        return "Context file";
    }
}

File: src/de/sciencecomputing/venuside/psi/ContextFile.kt
package de.sciencecomputing.venuside.psi

import de.sciencecomputing.venuside.ContextFileType
import de.sciencecomputing.venuside.ContextLanguage

import com.intellij.extapi.psi.PsiFileBase
import com.intellij.openapi.fileTypes.FileType
import com.intellij.psi.FileViewProvider

class ContextFile(viewProvider: FileViewProvider) :
        PsiFileBase(viewProvider, ContextLanguage) {
    override fun getFileType(): FileType = ContextFileType
    override fun toString(): String = "Context file"
}